### PR TITLE
Cloud docs

### PR DESCRIPTION
**Pull request by DailyDreaming**
Addresses #2123, #1972, #1967,  #1966. #1964, #1924, and #1886.

Other comments and changes addressed:
- Removed the duplicative "overview" section at the top of each heading.
- Reworked "cluster utilities".
- The "deploying a workflow" section has been changed and moved to the appendix, as it's basically an auto-deployment explanation section.
- Moved the CWL and WDL sections in running to after the sections on running the clouds, because right now they forward reference to the cloud sections.
- Reorganized the "running" section as:
-- Running in AWS
-- Running in Azure
-- Running in GCE
-- Running in HPC Environments
-- Deleted: Running in Openstack (this says nothing)
- Added options for toil itself and the launch-cluster utility.

Attempts to improve some of our current docs in terms of clarity and organization.  Feedback very welcome.

### CODE REVIEW

**Comment by bencvdb** [comment]
Clicking on Running Toil/ Running Toil in the Cloud opens itself as well as Running Toil/ Cloud Platforms.

Running Toil/ Cloud Platforms should be a subtopic of Running Toil/ Running Toil in the Cloud, maybe.

Running Toil/ Cloud Platforms could be renamed to Running Toil/ Supported Platforms.

Having a section called "Running Toil/ An Intro to Running Toil in the Cloud" is kind of weird when  there is also a "Running Toil/ Introduction"

**Comment by bencvdb** [comment]
Rename "Developing Toil Workflows/ Developing a Workflow" to something that reflects its contents.
EX:  Developing Toil Workflows/ Overview.

**Comment by bencvdb** [comment]
Regroup topics of "Developing Toil Workflows" 

//"API"
     Toil API
     Toil Job Utility API
     Job Store API
     Batch System API
// Job.*
//Exceptions

**Comment by bencvdb** [comment]
All of the subtopics of "Contributing to Toil" are on the same page. Is this intentional? Splitting them up make them more digestable.

**Comment by DailyDreaming** [comment]
I've regrouped and renamed some things.  Other things (like the current cloud topic structure) I am reluctant to change without the consensus of others, though I'd really like to change that too, there's been some opinion that it should not be changed.

And thanks!

File: docs/appendices/deploy.rst [hunk 1]
```diff
@@ -1,73 +1,32 @@
...
+little help on your part. We call this feature *auto-deployment* of a workflow.
 
+Let's first examine various scenarios of auto-deploying a workflow, which, as we'll see shortly cannot be
+auto-deployed. Lastly, we'll deal with the issue of declaring :ref:`Toil as a dependency <depending_on_toil>` of a
+workflow that is packaged as a setuptools distribution.
 
-Toil can be easily deployed to a remote host, given that both Python and Toil
-are present. The first order of business after copying your workflow to each
-host is to create and activate a virtualenv::
+Toil can be easily deployed to a remote host. First, assuming you've followed our `blog instructions`_ to install Toil
```
[the remark this answers is not in the document]
Comment (DailyDreaming) [reply docs/appendices/deploy.rst]: Good point.  I was kind of on the fence about it, but I think it's a good idea.  I'll add it in.

[the remark this answers is not in the document]
Comment (DailyDreaming) [reply docs/appendices/deploy.rst]: Done.

**Pull request merged**